src/main/java/com/uid2/admin/vertx/service/SharingService.java

```diff
@@ -150,6 +161,64 @@ private void handleSetKeyset(RoutingContext rc) {
         }
     }
 
+    // Method to check if one set contains any values from another set
+    private static <T> boolean containsAny(Set<T> set1, Set<T> set2) {
+        for (T element : set2) {
+            if (set1.contains(element)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void handleListAllKeysetsRelated(RoutingContext rc) {
+        try {
+            // Get value for site id
+            final Optional<Integer> siteIdOpt = RequestUtil.getSiteId(rc, "site_id");
+            if (!siteIdOpt.isPresent()) return;
+            final int siteId = siteIdOpt.get();
+
+            if (siteId != Const.Data.AdvertisingTokenSiteId && !SiteUtil.isValidSiteId(siteId)) {
```

Comment:
We can check if the site ID is present in the request as well? Right now we just return?